extensions-contrib/ambari-metrics-emitter/src/main/java/io/druid/emitter/ambari/metrics/AmbariMetricsEmitter.java

```diff
@@ -0,0 +1,233 @@
+          if (countLostEvents.getAndIncrement() % 1000 == 0) {
+            log.error(
+              "Lost total of [%s] events because of emitter queue is full. Please increase the capacity or/and the consumer frequency",
+              countLostEvents.get()
+            );
+          }
+        }
+      }
+      catch (InterruptedException e) {
+        log.error(e, "got interrupted with message [%s]", e.getMessage());
+        Thread.currentThread().interrupt();
+      }
+    } else if (event instanceof AlertEvent) {
+      for (Emitter emitter : emitterList) {
+        emitter.emit(event);
+      }
+    } else {
+      throw new ISE("unknown event type [%s]", event.getClass());
+    }
+  }
+
+  @Override
+  protected String getCollectorUri()
+  {
+    return collectorURI;
+  }
+
+  @Override
+  protected int getTimeoutSeconds()
+  {
+    return (int) (FLUSH_TIMEOUT / 1000);
+  }
+
+  private class ConsumerRunnable implements Runnable
+  {
+    @Override
+    public void run()
+    {
+      try {
+        int batchSize = config.getBatchSize();
+        TimelineMetrics metrics = new TimelineMetrics();
+        while (eventsQueue.size() > 0 && !exec.isShutdown()) {
+          try {
+            final TimelineMetric metricEvent = eventsQueue.poll(
+              config.getWaitForEventTime(),
+              TimeUnit.MILLISECONDS
+            );
+            if (metricEvent != null) {
+              metrics.addOrMergeTimelineMetric(metricEvent);
+              if (metrics.getMetrics().size() == batchSize) {
+                emitMetrics(metrics);
+                log.debug(
+                  "sent [%d] events",
+                  metrics.getMetrics().size()
+                );
+                metrics = new TimelineMetrics();
+              }
+            }
+          }
+          catch (InterruptedException e) {
+            log.error(e, e.getMessage());
+            Thread.currentThread().interrupt();
+          }
+        }
+        if (metrics.getMetrics().size() > 0) {
+          emitMetrics(metrics);
+          log.debug(
+            "sent [%d] events",
+            metrics.getMetrics().size()
+          );
+        }
+      }
+      catch (Exception e) {
+        log.error(e, e.getMessage());
+        if (e instanceof InterruptedException) {
+          Thread.currentThread().interrupt();
+        }
+      }
+
+    }
+  }
+
+  @Override
+  public void flush() throws IOException
+  {
+    if (started.get()) {
+      Future future = exec.schedule(new ConsumerRunnable(), 0, TimeUnit.MILLISECONDS);
+      try {
+        future.get(FLUSH_TIMEOUT, TimeUnit.MILLISECONDS);
+      }
+      catch (InterruptedException | ExecutionException | TimeoutException e) {
+        if (e instanceof InterruptedException) {
+          throw new RuntimeException("interrupted flushing elements from queue", e);
+        }
+        log.error(e, e.getMessage());
+      }
+    }
+  }
+
+  @Override
+  public void close() throws IOException
+  {
+    flush();
```

Comment:
need to make sure that only one call to flush is in the queue maybe synchronize around start ?